2-mandatory/3-financial-times.js

```diff
@@ -14,24 +20,48 @@ function potentialHeadlines(allArticleTitles) {
     (you can assume words will always be seperated by a space)
 */
 function titleWithFewestWords(allArticleTitles) {
-    // TODO
-}
+    let lengthOfTitles = []
+    for(let i in allArticleTitles){
+        lengthOfTitles.push(allArticleTitles[i].split(" ").length)
+    }
+    let minimumTitlesIndex = lengthOfTitles.indexOf(Math.min(...lengthOfTitles))
+    return allArticleTitles[minimumTitlesIndex]
+        
+    }
+
 
 /*
     The editor of the FT has realised that headlines which have numbers in them get more clicks!
     Implement the function below to return a new array containing all the headlines which contain a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
-}
+    let output = []
+    for(let i in allArticleTitles){
+        
+        if(allArticleTitles[i].match(/[0-9]/g) !== null){
```

Comment:
Great use of regular expressions, @Ekremteke actually simplified this with the `/d` operator. `/d` is a shortcut for `0-9` both are correct however I just felt it would be good to share that shortcut.

Reply:
That made sense, thank you.